contents/blog/5-ways-to-improve-analytics-data.md

```diff
@@ -0,0 +1,133 @@
+- iOS developer: Captures the same event but names it `user_sign_up`.
+- Android developer: Calls it `create account` but only captures it when they have finished signing up.
+
+Now, you must remember all these differences when looking at your data! Annoying! Using a naming convention prevents these problems. 
+
+### Suggested naming guide
+
+- Only use lowercase letters.
+
+- Use present-tense verbs, e.g., "submit" and "create" instead of "submitted" and "created".
+
+- Use snake case, i.e., `signup_button_click`
+
+- Create a dedicated list of allowed verbs, and don't deviate from it. Here's a list of the most common ones you're likely to use:
+  ```
+  - click
+  - submit
+  - create
+  - view
+  - add
+  - invite
+  - update
+  - delete
+  - remove
+  - start
+  - end
+  - cancel
+  - fail
+  - generate
+  - send
+  ```
+
+- For event names, use the `category:object_action` framework:
+
+  - `category` describes the context the event took place in – e.g., `account_settings` or `signup_flow` 
+
+  - `object` is a noun that refers to the component or location the event took place in, like `forget_password_button` or `pricing_page`.
+
+  - `action` is a verb that describes what happened, like `click`, `submitted`, or `create`
+
+  - Putting all of these together, you get `account_settings:forget_password_button_click` or `signup_flow:pricing_page_view` – The order is important as it makes it easier to find a specific event within a long list.
+
+- For property names:
+
+  - Use the `object_adjective` pattern. For instance: `user_id`, `item_price`, or `member_count`.
+
+  - Use `is/has` prefixes for boolean properties. For example: `is_subscribed`, `has_seen_upsell`.
+
+  - If the property's value is a date or timestamp, include _date or _timestamp at the end. For example, `user_creation_date`, `last_login_timestamp`.
+
+## 2. Filter out internal users
+
+A common mistake I see is developers inadvertently inflating their own metrics by not filtering out their own usage. This leads to massive bias in their data – especially in startups that have few users!
+
+For this reason, it's important to filter out events sent by your own team. There are a few ways to do this:
+
+- Filter out events from users with internal emails.
+- Add a property on your events `is_employee` or `is_test_user` and filter events where these are `true`.
+- Exclude internal IPs from your analytics tracking.
+- Filter events by domain host e.g., exclude `localhost:3000`, `staging.yourapp.com`.
+- Turn off tracking in development using a `localhost`, `dev`, or config check.
+
+> **Tip:** Rather than add these filters to each new [insight](/docs/product-analytics/insights) you create, set it as your default filter.
+> 
+> For a detailed guide on how to do this in PostHog, see the tutorial on [how to filter out internal users in PostHog](/tutorials/filter-internal-users).
+
+## 3. Prefer backend to frontend tracking
```

Comment:
Should we add something in this section about:

1. The backend libraries we support / setup guides for backend tracking in PostHog
2. How to label backend events vs frontend events in PostHog? 

On the latter, I assume that's a use case Data Management and applying tags to events to show where they were captured?

Reply:
On both points: We have 2 callouts already (out of 5 points), so I'm reluctant to add more posthog specific information, in order to avoid pushing posthog too hard in this post

> How to label backend events vs frontend events in PostHog?

There are a few ways to do this in PostHog e.g. posthog logs a property from which library the event came from